### ISSUE TITLE
ZFS module does not compile with 6.2.8_1

### ISSUE THREAD

**Issue by gtryujb** [void-linux/void-packages#43013]
### Is this a new report?

Yes

### System Info

Void 6.1.21_1 x86_64 GenuineIntel uptodate rrrrmmnFFFFFF

### Package(s) Affected

zfs-2.1.9_1 linux6.2-6.2.8_1

### Expected behaviour

The module compiles and installs.

### Actual behaviour

The ZFS module fails to compile against 6.2.8_1.

`xbps-reconfigure -f linux6.2`:
```
linux6.2: configuring ...
Executing post-install kernel hook: 10-dkms ...
Available DKMS module: nvidia-525.89.02.
Available DKMS module: zfs-2.1.9.
Building DKMS module: nvidia-525.89.02... done.
Building DKMS module: zfs-2.1.9... FAILED!
Generating kernel module dependency lists... done.
Executing post-install kernel hook: 20-initramfs ...
dracut-install: Failed to find module 'zfs'
dracut: FAILED:  /usr/lib/dracut/dracut-install -D /var/tmp/dracut.b8v8th/initramfs --kerneldir /lib/modules/6.2.8_1/ -m zfs
dracut: installkernel failed in module zfs
Executing post-install kernel hook: 50-efibootmgr ...
Executing post-install kernel hook: 50-grub ...
Generating grub configuration file ...
Found linux image: /boot/vmlinuz-6.2.8_1
Found linux image: /boot/vmlinuz-6.1.21_1
Found initrd image: /boot/initramfs-6.1.21_1.img
Found linux image: /boot/vmlinuz-6.1.19_1
Found initrd image: /boot/initramfs-6.1.19_1.img
Found linux image: /boot/vmlinuz-5.19.17_1
Found initrd image: /boot/initramfs-5.19.17_1.img
Warning: os-prober will not be executed to detect other bootable partitions.
Systems on them will not be added to the GRUB boot configuration.
Check GRUB_DISABLE_OS_PROBER documentation entry.
Adding boot menu entry for UEFI Firmware Settings ...
done
linux6.2: configured successfully.
```

End of `/var/lib/dkms/zfs/2.1.9/build/make.log`:
```
  LD [M]  /var/lib/dkms/zfs/2.1.9/build/module/zstd/zzstd.o  
  LD [M]  /var/lib/dkms/zfs/2.1.9/build/module/zfs/zfs.o
  MODPOST /var/lib/dkms/zfs/2.1.9/build/module/Module.symvers
ERROR: modpost: GPL-incompatible module zfs.ko uses GPL-only symbol 'bio_start_io_acct'
ERROR: modpost: GPL-incompatible module zfs.ko uses GPL-only symbol 'bio_end_io_acct_remapped'
make[2]: *** [scripts/Makefile.modpost:138: /var/lib/dkms/zfs/2.1.9/build/module/Module.symvers] Error 1
make[1]: *** [Makefile:1973: modpost] Error 2
make[1]: Leaving directory '/usr/src/kernel-headers-6.2.8_1'
make: *** [Makefile:55: modules-Linux] Error 2
make: Leaving directory '/var/lib/dkms/zfs/2.1.9/build/module'
```

### Steps to reproduce

1. `xbps-reconfigure -f linux6.2`

**Comment by classabbyamp** [comment]
I don't think zfs supports 6.2 yet

**Comment by adrian-bl** [comment]
> I don't think zfs supports 6.2 yet

ZFS worked well up until and including 6.2.7 - that's a breakage in 6.2.8+

The related ZOL issue is at https://github.com/openzfs/zfs/issues/14658 and https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=5f27571382ca42daa3e3d40d1b252bf18c2b61d2 seems to be the commit which broke it.

The fix on the ZOL side seems to be https://github.com/openzfs/zfs/pull/14677 which is likely included in 2.1.10

**Comment by ahesford** [comment]
If we decide to bump `linux` to point to `linux6.2` before ZFS has a release that includes that patch, we may pull the patch locally. In the meantime, failure to build on `linux6.2` is not unexpected and we can't assume successful builds on prior releases in the series imply ongoing functionality. This is not a Void bug.
